_posts/2011-07-17-testing-legacy-hard-wired-dependencies.md

```diff
@@ -14,44 +14,44 @@ problems. The most common one is related to hard-wired dependencies -
 
 Let's look at this piece of code:
 
-{% highlight java %}
+```
 public List<Trip> getTripsByUser(User user) throws UserNotLoggedInException {
-        List<Trip> tripList = new ArrayList<Trip>();
-        User loggedUser = UserSession.getInstance().getLoggedUser();
-        boolean isFriend = false;
-        if (loggedUser != null) {
-            for (User friend : user.getFriends()) {
-                if (friend.equals(loggedUser)) {
-                    isFriend = true;
-                    break;
-                }
+    List<Trip> tripList = new ArrayList<Trip>();
+    User loggedUser = UserSession.getInstance().getLoggedUser();
+    boolean isFriend = false;
+    if (loggedUser != null) {
+        for (User friend : user.getFriends()) {
+            if (friend.equals(loggedUser)) {
+                isFriend = true;
+                break;
             }
-            if (isFriend) {
-                tripList = TripDAO.findTripsByUser(user);
-            }
-            return tripList;
-        } else {
-            throw new UserNotLoggedInException();
         }
+        if (isFriend) {
+            tripList = TripDAO.findTripsByUser(user);
+        }
+        return tripList;
+    } else {
+        throw new UserNotLoggedInException();
     }
-{% endhighlight %}
+}
+```
 
 Horrendous, isn't it? The code above has loads of problems, but before
 we change it, we need to have it covered by tests.
 
 There are two challenges when unit testing the method above. They are:
 
 
-{% highlight java %}
+```
 User loggedUser = UserSession.getInstance().getLoggedUser(); // Line 3
 tripList = TripDAO.findTripsByUser(user);                    // Line 13
-{% endhighlight %}
+```
 
 As we know, unit tests should test just one class and not its
 dependencies. That means that we need to find a way to mock the
 Singleton and the static call. In general we do that injecting the
 dependencies, but we have a
-[rule](http://craftedsw.blogspot.com/2011/07/working-with-legacy-code.html),
+[rule]({{site.baseurl}}/2011/07/03/working-with-legacy-code),
```

Comment:
Hey sandro, the baseurl is only for links that are hosted inside our domain. In this case the link was fine. 

You can check the preview here: https://codurance-site-pr.s3.amazonaws.com/site-fixing-old-legacy-code-blog/2011/07/16/testing-legacy-hard-wired-dependencies/index.html